### PR TITLE
fix: UI inconsistencies in Datasource card and page

### LINKED ISSUE #13933: [Bug]: Mismatch with design in Datasource page
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Sample database component is not as per design - https://www.figma.com/file/i80YbQLLs5mXthjIHclPpX/Appsmith-Design-System?node-id=4966%3A27307

1. > [Helper text alignment](https://www.notion.so/appsmith/870d3c54354d460f9a90457ce6d7c98d?v=b6dd8044acd346c2a55c9d2b925bc454&p=e90e85c6e6a9495aa87ae46fb2b736e8)
2. >[Hover state](https://www.notion.so/appsmith/870d3c54354d460f9a90457ce6d7c98d?v=b6dd8044acd346c2a55c9d2b925bc454&p=78a83688424c4e84b80c7cedf20ed911)
3. >[Spacing between icon and datasource name](https://www.notion.so/appsmith/870d3c54354d460f9a90457ce6d7c98d?v=b6dd8044acd346c2a55c9d2b925bc454&p=fc4435a89a184e8ea84b303c0205dd9c)

### Steps To Reproduce

1. Click on add new datasource or show all datasources

### Public Sample App

_No response_

### Version

Cloud

### LINKED ISSUE #13936: [Bug]: Datasource card design mismatch
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Card design doesn't match with the figma - https://www.figma.com/file/i80YbQLLs5mXthjIHclPpX/Appsmith-Design-System?node-id=4966%3A27307

1. [Vertical space mismatch](https://www.notion.so/appsmith/870d3c54354d460f9a90457ce6d7c98d?v=b6dd8044acd346c2a55c9d2b925bc454&p=0c3ca02269404483991822793c6c8b57)
2. [Update show more icon](https://www.notion.so/appsmith/870d3c54354d460f9a90457ce6d7c98d?v=b6dd8044acd346c2a55c9d2b925bc454&p=ad1603b3ba704378bec3dbc21f899b12)
3.  [Inconsistent spacing between the icon and the datasource name](https://www.notion.so/appsmith/870d3c54354d460f9a90457ce6d7c98d?v=b6dd8044acd346c2a55c9d2b925bc454&p=fc4435a89a184e8ea84b303c0205dd9c)

### Steps To Reproduce

1. Click on show all datasources

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by ChandanBalajiBP**
## Description

> Inconsistency in design of the following

1. Vertical spacing mismatch between text in datasource card
2. Icon updated in show more collapse button
3. Spacing between the icon and the datasource name in datasource card.
4. Helper text alignment in mock datasource card
5. Hover state in Mock datasource, New API and Databases cards.
6. Spacing issues with section title & first line-item in Datasources pane 
7. Spacing inconsistency in Datasources page(Have fixed most of which i could identify)
8. Datasource title font size mismatch
9. Font size and color mismatch in all the datasource page
10. Hover state on datasource icon

Fixes #13933 
Fixes #13936

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: Fix/Mismatch-with-design-in-Sample-Datasource-Card 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.7 **(0)** | 38.66 **(0)** | 36.19 **(0.01)** | 56.96 **(0)**
 :green_circle: | app/client/src/components/ads/Button.tsx | 92.36 **(0)** | 82.76 **(0.86)** | 97.78 **(0)** | 92.96 **(0)**
 :green_circle: | app/client/src/components/ads/Tabs.tsx | 22.34 **(0.83)** | 9.43 **(0)** | 0 **(0)** | 23.53 **(0.91)**
 :green_circle: | app/client/src/components/utils/CollapseComponent.tsx | 42.86 **(2.86)** | 20 **(0)** | 0 **(0)** | 50 **(3.85)**
 :green_circle: | app/client/src/pages/Editor/IntegrationEditor/DatasourceCard.tsx | 73.49 **(0.32)** | 69.44 **(0.87)** | 14.29 **(0)** | 72.84 **(0.34)**
 :green_circle: | app/client/src/pages/Editor/IntegrationEditor/MockDataSources.tsx | 50 **(4.05)** | 88.89 **(1.39)** | 0 **(0)** | 50 **(4.05)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

File: app/client/src/components/ads/Tabs.tsx [hunk 1]
```diff
@@ -134,10 +135,10 @@ const TabsWrapper = styled.div<{
 `;
 
 export const TabTitle = styled.span<{ responseViewer?: boolean }>`
-  font-size: ${(props) => props.theme.typography.h5.fontSize}px;
-  font-weight: ${(props) => props.theme.typography.h5.fontWeight};
-  line-height: ${(props) => props.theme.typography.h5.lineHeight - 3}px;
-  letter-spacing: ${(props) => props.theme.typography.h5.letterSpacing}px;
+  font-size: ${(props) => props.theme.typography.h4.fontSize}px;
+  font-weight: ${(props) => props.theme.typography.h4.fontWeight - 100};
```
Comment (Irongade) [inline app/client/src/components/ads/Tabs.tsx:139]: Why are we subtracting 100 from the h4 fontWeight here?

Comment (Irongade) [reply app/client/src/components/ads/Tabs.tsx:139]: You can consider creating a list of or checking for fontWeight values (outside of the font size spec) in the theme object. Then using it directly here. 

Possible schema of values:

- 100 – Thin.
- 200 – Extra Light (Ultra Light)
- 300 – Light.
- 400 – Normal.
- 500 – Medium.
- 600 – Semi Bold (Demi Bold)
- 700 – Bold.
- 800 – Extra Bold (Ultra Bold)

Comment (ChandanBalajiBP) [reply app/client/src/components/ads/Tabs.tsx:139]: ok, i will update. Would you suggest any other change?? (So that i can update that also.)

Comment (ChandanBalajiBP) [reply app/client/src/components/ads/Tabs.tsx:139]: Okay, will update these changes.

**Review by rashmi-rai — APPROVED** [review]

**Review by Irongade — APPROVED** [review]

**Review by Irongade — DISMISSED** [review]

**Comment by prapullac** [comment]
Test Plan link 
- [ ] [Test Cases] : Mismatch with design in Sample Datasource Card (https://github.com/appsmithorg/TestSmith/issues/1733)

**Review by Irongade — APPROVED** [review]

**Review by Parthvi12 — COMMENTED** [review]
Just a FYI, for UI inconsistencies we can use snapshot tests as well, which takes snapshot while running tests and compare pixel by pixel with that of reference image. Otherwise this LGTM.

**Comment by btsgh** [comment]
Tested on the DP with the help of figma/notion docs, discussed with @ChandanBalajiBP - the changes that were made are reflecting.

**Pull request merged**